driver/network_driver.go

```diff
@@ -61,6 +62,11 @@ func (d NetworkDriver) GetCapabilities() (*network.CapabilitiesResponse, error)
 func (d NetworkDriver) CreateNetwork(request *network.CreateNetworkRequest) error {
 	logutils.JSONMessage("CreateNetwork", request)
 
+	knownOpts := map[string]bool{
+		"com.docker.network.generic":     true,
+		"com.docker.network.internal":    true,
+		"com.docker.network.enable_ipv6": true}
+
 	genericOpts, ok := request.Options["com.docker.network.generic"]
```

Comment:
It returns an error to the user if any com.docker.network.generic options are supplied. Your new code ignores that option and returns a slightly different error when different options are supplied. It's wrong to have two different bits of code doing more or less the same thing.